core/data/src/main/java/com/pronoidsoftware/core/data/di/UserInitialsModule.kt

```diff
@@ -0,0 +1,23 @@
+package com.pronoidsoftware.core.data.di
+
+import com.pronoidsoftware.core.domain.SessionStorage
+import com.pronoidsoftware.core.domain.util.initializeAndCapitalize
+import dagger.Module
+import dagger.Provides
+import dagger.hilt.InstallIn
+import dagger.hilt.components.SingletonComponent
+import javax.inject.Named
+import kotlinx.coroutines.runBlocking
+
+@Module
+@InstallIn(SingletonComponent::class)
+object UserInitialsModule {
+
+    @Provides
+    @Named("User Initials")
+    fun provideUserInitials(sessionStorage: SessionStorage): String? {
+        return runBlocking {
+            sessionStorage.get()?.fullName?.initializeAndCapitalize()
+        }
+    }
+}
```

Comment:
This is something I wouldn't inject, since I would never expect this piece of logic in a Dagger module. If someone else would start working on the app and also had to add the initials elsewhere, I'd be 99% sure they would do it manually without using this injected instance 😄 
To keep things clear, I'd just inject the session storage and process it where it's needed.

Reply:
Makes sense, I've reverted it 👍